airflow/providers/microsoft/azure/hooks/adls_v2.py

```diff
@@ -0,0 +1,307 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Any
+
+from azure.core.exceptions import ResourceExistsError, ResourceNotFoundError
+from azure.identity import ClientSecretCredential
+from azure.storage.filedatalake import (
+    DataLakeDirectoryClient,
+    DataLakeFileClient,
+    DataLakeServiceClient,
+    DirectoryProperties,
+    FileSystemClient,
+    FileSystemProperties,
+)
+from hooks.base import BaseHook
+
+
+class AdlsClientHook(BaseHook):
+    """
+    This Hook interacts with ADLS gen2 storage account it mainly helps to create and manage
+    directories and files in storage accounts that have a hierarchical namespace. Using Alds_v2 connection
+    details create DataLakeServiceClient object
+
+    Due to Wasb is marked as legacy and and retirement of the (ADLS1) it would be nice to
+    implement ALDS gen2 hook for interacting with the storage account.
+
+    .. seealso::
+        https://learn.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-directory-file-acl-python
+
+    :param adls_v2_conn_id: Reference to the :ref:`adls_v2 connection <howto/connection:adls_v2>`.
```

Comment:
Hi @bharanidharan14 , it would be great if we could rename things as follows:

* `airflow/providers/microsoft/azure/hooks/adls_v2.py` -> `airflow/providers/microsoft/azure/hooks/azure_data_lake.py` (use existing module)
* `AdlsClientHook` -> AzureDataLakeStorageV2Hook` (be more explicit since we'll have V1 and V2 ADLS hooks coexisting for some time in Airflow/Azure, it is probably worth having V2 explicit in the class name)
* `adls_v2_conn_id` -> `adls_conn_id` (if the class name has V2, we can document that this parameter should be a V2 connection ID, but I think the parameter could be called only adls_conn_id)

Reply:
@tatiana Will address this and let you know

Reply:
@tatiana Addressed your suggestion

Reply:
@tatiana Can you review it now